robot-agent/packageMonitor.js

```diff
@@ -0,0 +1,99 @@
+const fs = require('fs');
+
+const constants = require('./constants');
+const { spawn, exec, execSync } = require('child_process');
+const { EventEmitter } = require('events');
+
+const { getLogger } = require('@transitive-sdk/utils');
+const log = getLogger('packageMonitor.js');
+log.setLevel('info');
+
+class PackageMonitor extends EventEmitter {
+  constructor() {
+    super();
+    this.installedPackages = {};
+  }
+  async initialize() {
+    await this.updateAllInstalledPackages();
+  }
+  async updateAllInstalledPackages() {
+    log.debug('Updating installed packages...');
+    const { getInstalledPackages } = require('./utils');
+    const installedPackages = getInstalledPackages();
+    const promises = installedPackages.map(async (pkgName) => {      
+      return this.refreshSinglePackage(pkgName, installedPackages);
+    });
+    await Promise.all(promises);
+    log.debug('Installed packages updated:', this.installedPackages);
+  }
+  async refreshSinglePackage(pkgName, currentInstalledPackages = null) {
```

Comment:
Given our discussion about using `runningPackages` directly, I don't know that we even need this function.